gtsam/hybrid/HybridBayesNet.cpp

```diff
@@ -43,115 +42,60 @@ bool HybridBayesNet::equals(const This &bn, double tol) const {
 }
 
 /* ************************************************************************* */
-// The implementation is: build the entire joint into one factor and then prune.
-// TODO(Frank): This can be quite expensive *unless* the factors have already
-// been pruned before. Another, possibly faster approach is branch and bound
-// search to find the K-best leaves and then create a single pruned conditional.
 HybridBayesNet HybridBayesNet::prune(
-    size_t maxNrLeaves, const std::optional<double> &deadModeThreshold) const {
+    size_t maxNrLeaves, const std::optional<double> &marginalThreshold,
```

Comment:
I am not a fan of `marginalThreshold` since it doesn't indicate which marginals we're talking about (discrete? continuous?), and it seems to indicate we already have marginals which we wish to threshold on, not compute marginals.
Better name please?